server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -280,38 +280,42 @@ Fate<Manager> fate() {
       /* UNLOAD_ROOT_TABLET */      {O, O, O, X, X, X, X},
       /* STOP */                    {O, O, O, O, O, X, X}};
   //@formatter:on
-  synchronized void setManagerState(ManagerState newState) {
-    if (state.equals(newState)) {
+  synchronized void setManagerState(final ManagerState newState) {
+    if (state == newState) {
       return;
     }
     if (!transitionOK[state.ordinal()][newState.ordinal()]) {
       log.error("Programmer error: manager should not transition from {} to {}", state, newState);
```

Comment:
Not requesting any change, this is just a question.  Would anyone happen to know why this does not throw an exception instead of or in addition to logging?

Reply:
Good question. I checked the history, and it looks like this behavior has been the same since the initial commit at the ASF. I could try to investigate the history prior to the software grant agreement, but it would take time and I doubt I'll find any explanation or documentation to explain why.

Reply:
@keith-turner This made sense to change to a thrown exception. I don't think it'd be safe to merely log an error. Even though I've never actually seen this error... if it happened, I think I'd prefer to see the stack trace and the state change to be prevented from occurring, rather than just see the error message and the state change occur anyway.